packages/flutter/lib/src/material/segmented_button.dart

```diff
@@ -0,0 +1,726 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+import 'button_style.dart';
+import 'color_scheme.dart';
+import 'colors.dart';
+import 'icons.dart';
+import 'material.dart';
+import 'material_state.dart';
+import 'segmented_button_theme.dart';
+import 'text_button.dart';
+import 'theme.dart';
+
+/// Data describing a segment of a [SegmentedButton].
+class ButtonSegment<T> {
+  /// Construct a SegmentData
+  ///
+  /// One of [icon] or [label] must be non-null.
```

Comment:
This seems a little odd, since they're both Widgets.  Is it because of different default styles, or needing to apply different styles?

Reply:
Are you commenting on the fact that they are separate widgets, or that one of them must be non-null? 

The reason they are separate is that when an item with both an icon and a label is selected, the icon is replaced with a checkbox (see the second row of the image above). In order to do this, they need to be distinct widgets.

Reply:
I think I misread the comment.  I was thinking it was saying that only one of them could be set, and since they were both widgets, I wondered why it wasn't just one value. I see now that both can be set.